app/README.md

```diff
@@ -23,3 +31,35 @@ running `yarn start` starts a local development server which automatically recom
 the development server runs on port 3000 and forwards any unhandled request to the `"proxy"` defined in the `package.json` file. starting the `maiden` server in a separate shell allows api calls to be handled during development.
 
 installing both the `reactjs` and `redux` devtools browser extensions is highly recommended. 
+
+## repl connection
+
+the `maiden` repl uses websockets to directly connect to the `matron` (lua) and `crone` (supercollider) consoles bypassing the http backend. in order to determine which host/ip and socket to connect to `maiden` reads the `repl-endpoints.json` file (`app/public/repl-endpoints.json`) from the http backend.
+
+when running both the frontend and backend locally during development it is still possible to make the repl work by locally editing `repl-endpoints.json`, the default configuration looks like:
+
+```
+{
+    "matron": "ws://maiden_app_location:5555",
+    "sc": "ws://maiden_app_location:5556"
+}
+```
+
+the `maiden_app_location` is a special value which causes the frontend to attempt to connect to whichever host/ip `maiden` itself was loaded from. when running the frontend/backend locally one change the config to point specifically a the hostname or ip addr of the norns device. for example:
+
```

Comment:
maybe replace

> "when running the frontend/backend locally one change" 

with

"when running the frontend/backend locally one can change"

or 

"when running the frontend/backend locally, change"

?